### PR TITLE
Add troubleshooting section to manual

### PR DESCRIPTION
**Pull request by msimberg**
Adds a short troubleshooting section to the manual. Generally I think we should try to keep tips like these in their appropriate sections, but some things pop up frequently enough that I think it's worth having a separate section just for common problems. Doesn't contain much yet, but it'll hopefully expand over time.

### CODE REVIEW

**Comment by hkaiser** [comment]
FWIW, @McKillroy has started an HPX FAQ documentation (https://github.com/McKillroy/hpx_snippets), where this kind of information could go.

File: docs/sphinx/manual/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Boost.ProgramOptions is not ABI compatible between all C++ versions and compilers.
+Because of this you may see linker errors similar to this:
+
+.. code-block:: text
+
+   ...: undefined reference to `boost::program_options::operator<<(std::ostream&, boost::program_options::options_description const&)'
+
+if you are not linking to a compatible version of Boost.ProgramOptions. We
+recommend that you always build |boost|_ libraries and |hpx| with the same
+compiler and C++ standard.
```
Comment (hkaiser) [inline docs/sphinx/manual/troubleshooting.rst]: Update this with regards to the new `hpx::program_options` module?

Comment (msimberg) [reply docs/sphinx/manual/troubleshooting.rst]: Yes, my memory is indeed short...

Comment (msimberg) [reply docs/sphinx/manual/troubleshooting.rst]: Done.

**Comment by msimberg** [comment]
> FWIW, @McKillroy has started an HPX FAQ documentation (https://github.com/McKillroy/hpx_snippets), where this kind of information could go.

Yeah, that's really nice indeed. But long term we want to have these kind of things in our documentation, no? We shouldn't need to rely on @McKillroy maintaining this.

**Comment by McKillroy** [comment]
Just to clarify - my idea was just to have something I could quickly and easily work on, with a positive side effect you could use it and integrate what looks useful to you. 
I totally don't want to create some extra "grey" HPX documentation. 
The best effect could be that some of my stuff simply become superfluous, because it's all covered in the official documentation.

**Pull request merged**